### ISSUE TITLE
[3.3.5][6.x][Hunter] Scatter Shot visual bug in moviment targets.

### ISSUE THREAD

**Issue by Keader** [TrinityCore/TrinityCore#16155]
If you use Scatter Shot in moviment targets, he "teleport" in desorientation moviment.

Old Commit has fixed that, dont work anymore (https://github.com/TrinityCore/TrinityCore/pull/7975)

Videos of servers based at Trinity

https://www.youtube.com/watch?v=28aRmHO-WpQ
https://www.youtube.com/watch?v=pSDQ5yZGmx4

How should work : https://www.youtube.com/watch?time_continue=119&v=PVhwiHmV1BI
(Even as a private server based on TC, is there working properly, i dont have time for search vid on retail)

Tc Rev 8a0bbc386982612077e6f83f1816b02be757e371

Old Issues: #2689 #2013 #466 #1011

**Comment by Rushor** [comment]
old fix for this was https://github.com/TrinityCore/TrinityCore/commit/3d644d8d119b9009f10ea637c52a01dad32708c9

but this no longer works

**Comment by Goatform** [comment]
Similar issue with Fear movement.

**Comment by Re3os** [comment]
https://github.com/TrinityCore/TrinityCore/issues/15952

**Comment by Keader** [comment]
Your issue is confusing o.O
Sotimes is about fear, somtimes about scatter lol

**Comment by tkrokli** [comment]
Isn't both Fear and Scatter Shot linked by using `ConfusedMovementGenerator` on its target?

**Comment by Goatform** [comment]
It might be. But I think that Fear is using <code>FleeingMovementGenerator</code>

**Comment by tkrokli** [comment]
OK, your coding experience is better than mine. I just based my guess on what little I can remember.

**Comment by Keader** [comment]
Same problem happens with the Blind, spell of rogues.
If when using the spell the target is stopped, works normally.
However if it is in motion happens visual bug.
In the case of the Blind "teleport" is less noticeable because the character "walks" and not "run" while confusing.

**Comment by Goatform** [comment]
@Keader, test this patch, it's based on 3d644d8 commit: 
``` diff
Author: Goatform
Based on: 3d644d8

@@ -1450,6 +1450,12 @@ uint8 Player::GetChatFlags() const
 
 bool Player::TeleportTo(uint32 mapid, float x, float y, float z, float orientation, uint32 options)
 {
+    if (HasUnitState(UNIT_STATE_LOST_CONTROL))
+    {
+        StopMoving();
+        GetMotionMaster()->Clear();
+    }
+
     if (!MapManager::IsValidMapCoord(mapid, x, y, z, orientation))
     {

**Comment by Demonid** [comment]
@Goatform I been using that code for years now, but it doesn't work for spells, it's for the teleport that players make after ending a Battleground, Arena etc.

**Comment by Goatform** [comment]
Aham okay, thanks for feedback @Demonid! I'll do some research related to this issue.

**Comment by Rushor** [comment]
i only seems to be visual - the target which gets the scatter shot will not move, only the hunter which casts the scatter shot will see this bug

**Comment by Goatform** [comment]
It is just visual, but it's very bad thing for PvP and sometimes it could confuse player(s).

**Comment by Demonid** [comment]
@Goatform @Rushor Yes, and this applies to most spells that do any kind of movement, scatter, fear, even charge, intercept and death grip.

This has been bugged since at least 2011, hope someday this get fixed, it's useful not only to 3.3.5 but to 6.x.x (I guess this bug applies to 6.xx)

**Comment by Rushor** [comment]
general movement issue. combined to the (now updated issue) https://github.com/TrinityCore/TrinityCore/issues/15952
